components/rating/stories/template.js

```diff
@@ -13,30 +13,78 @@ export const Template = ({
   rootClass = "spectrum-Rating",
   size = "m",
   max = 5,
+  isReadOnly = false,
   customClasses = [],
   id,
   ...globals
 }) => {
+  function setFocus(rating, focused) {
+    rating.classList[focused ? 'add' : 'remove']('is-focused');
+  }
+
+  function setValue(rating, value) {
+    const input = rating.querySelector(`.${rootClass}-input`);
+    input.value = value;
+
+    rating.querySelectorAll(`.${rootClass}-icon`)?.forEach((el, idx) => {
+        el.classList[idx <= value - 1 ? 'add' : 'remove']('is-selected');
+        el.classList[idx === value - 1 ? 'add' : 'remove']('is-currentValue');
+    });
+  }
+
   return html`
     <div
       class=${classMap({
         [rootClass]: true,
         [`${rootClass}--size${size?.toUpperCase()}`]: typeof size !== "undefined",
         ...customClasses.reduce((a, c) => ({ ...a, [c]: true }), {}),
       })}
-      id=${ifDefined(id)}>
+      id=${ifDefined(id)}
+      @focusin=${(e) => {
+        const rating = ended.target.closest(`.${rootClass}`);
```

Comment:
If we can separate focusin and focusout logic in a separate function that would make this more cleaner